### PR TITLE
users: Show loading state

### PR DESCRIPTION
**Pull request by marusak**
Current implementation would show blank page and wait until everything was loaded. Now the overview is shown with loading state inside groups and users views. Details page shows big spinner.

Screenshot from 2022-12-09 13-08-21
Screenshot from 2022-12-09 13-08-08

### CODE REVIEW

File: pkg/lib/cockpit-components-table.jsx [hunk 1]
```diff
@@ -133,6 +135,13 @@ export const ListingTable = ({
             : null
     );
 
+    if (loading)
+        return <EmptyState>
+            <EmptyStateBody>
+                <div>{loading}</div>
```
Comment (KKoukiou) [inline pkg/lib/cockpit-components-table.jsx:141]: I bet the div is not needed.

File: pkg/users/account-details.js [hunk 2]
```diff
@@ -173,6 +174,17 @@ export function AccountDetails({ accounts, groups, shadow, current_user, user })
                 .catch(show_unexpected_error);
     }
 
+    if (!accounts.length) {
+        return (
+            <EmptyState variant={EmptyStateVariant.small}>
+                <Spinner className="ct-spinner" isSVG size="xl" />
```
Comment (KKoukiou) [inline pkg/users/account-details.js:180]: THis class is not defined not used.

**Review by tomasmatus — APPROVED** [review]
Looks like the tests are failing on the flaky check-account locked but you're looking into that on another branch. The loading state looks good!

**Comment by garrett** [comment]
Technically, loading should look different:

https://www.patternfly.org/v4/components/table/html-demos/loading-state-demo/

image

But having some sort of loading indicator is definitely better than none. And we'd probably want to switch inline table/list loading indicators across Cockpit.

---

So, I'm fine with landing this as-is and doing some follow-ups. Or, we could have a fixup here and follow-ups for others.

**Review by KKoukiou — APPROVED** [review]

**Comment by martinpitt** [comment]
[two](https://cockpit-logs.us-east-1.linodeobjects.com/pull-18015-20221212-113219-7e3c68d1-fedora-36/log.html#317-1) [TestAccounts](https://cockpit-logs.us-east-1.linodeobjects.com/pull-18015-20221212-113215-7e3c68d1-rhel-8-7/log.html#317) failures due to affected retry. Trying again.

**Pull request merged**